### ISSUE TITLE
Mania hold notes stack on top of other notes

### ISSUE THREAD

**Issue by Bruno5430** [ppy/osu#22272]
### Type

Cosmetic

### Bug description

While it was mentioned in #21994 it's mentioning only the argon skin also it doesn't point how it behaves compared to stable.

In stable, hold notes (body and tails) stack underneath the next notes.
This is not the case in lazer, as hold notes stack on top of the next notes, making LN maps even harder to read.

### Screenshots or videos

| Lazer: (445ms scroll speed) | Stable: (30 scroll speed) |
| --- | --- |
| [Beatmap: Camellia - Nightmare city](https://osu.ppy.sh/beatmapsets/1588619#mania/3244597) |
| ![osu_2023-01-17_23-57-29](https://user-images.githubusercontent.com/111712756/213072066-894bd200-373d-4ed9-945c-6f8b0fa05a25.png) | ![screenshot141](https://user-images.githubusercontent.com/111712756/213071803-b71e4220-1ff4-4751-b6ec-038efbc8a673.jpg) |

### Version

2022.1228.0

### Logs

-

**Comment by peppy** [comment]
I'm pretty sure the left is more correct? At least to my eye.

**Comment by Trung0246** [comment]
Are these behaviors legible for a separate visual option? Guess some people like the left version while some would like the right version.

**Comment by Bruno5430** [comment]
Sorry for the late response, I've been busy lastly

First of all, I wanna point out the trend with hold notes in osu!mania legacy skins to understand better user-expectations:
There are 4 types of elements commonly present in legacy skins for hold note ~~tails~~ ends (still trying to memorize the new standardized names for objects)
| Bar end | Tail end |
| :---: | :---: |
| Used on the "argon" skin and old bar skins, like the osu! classic skin, or Beatmania styled skins, were they use a similar or the same texture for the start and the end of the hold note | Very common, the tail starts at the end of the body |
| ![bar_classic](https://user-images.githubusercontent.com/111712756/213557930-f4dc01a8-9c4f-460f-ad74-e8c3e5e619ff.png) ![bar_argon](https://user-images.githubusercontent.com/111712756/213558338-d325c2e6-81d5-46f7-b693-7b2af325057c.png) | ![noteTG](https://user-images.githubusercontent.com/111712756/213594058-36deeda4-7036-4c2f-9ddc-277dcb605720.png) ![noteT2](https://user-images.githubusercontent.com/111712756/213557620-f270e128-323f-4dd4-b066-527eaab1c8c1.png) |
| ![classic](https://user-images.githubusercontent.com/111712756/213554069-db62b450-1b46-4af6-9a6a-d643f7d7a90f.png)  ![argon](https://user-images.githubusercontent.com/111712756/213554736-2723bded-3bc2-4a60-be8c-85e8128951e8.png) | ![tail_tri](https://user-images.githubusercontent.com/111712756/213590661-c293131b-1a78-4284-ae8d-212106440a6c.png) ![tail_orb](https://user-images.githubusercontent.com/111712756/213592218-69c0d4ae-6a61-48b7-ae02-74bed1039612.png) |

| Non-overlapping tail end | Percy/Cut LN |
| :---: | :---: |
| (I didn't find it's real name, I made it for demonstration) Common on [remixed skins](https://osu.ppy.sh/community/forums/124), the tail ends at the same point as the body of the hold note, making it unlikely to overlap with the next object. It's done by overlapping a mask with the color of the playfield on top of the body of the hold note (same concept but not equal behavior to percy/cut LNs) | Very controversial (being discussed in #20576 and #12057), the tail can end at the same point __or before__ the end of the body of the hold note, this is possible because the actual tail is embedded with the texture of the body, leading to a 40.000px long texture |
| ![noteTG](https://user-images.githubusercontent.com/111712756/213595097-479a45b9-aaf2-44e6-854f-f72134963fc1.png) ![noteT](https://user-images.githubusercontent.com/111712756/213594779-cdfaafb2-316f-4ecb-93d8-6059f76b54e6.png) | ![noteLG](https://user-images.githubusercontent.com/111712756/213596276-b7de4533-0198-4bbf-9824-7759332257e1.png) ![holdbody](https://user-images.githubusercontent.com/111712756/213596934-dc894e46-0afc-422f-925c-9ce997d2c69e.png) Textures were cut for preview | 
| ![tail_tri_non](https://user-images.githubusercontent.com/111712756/213593337-e63bfdf7-3ac0-4d28-b741-be5bec98f249.png) ![tail_orb_non](https://user-images.githubusercontent.com/111712756/213591367-2825a775-e4c3-496c-a2ea-4f1b680491de.png) | ![percy_tri](https://user-images.githubusercontent.com/111712756/213612629-3023c65d-371b-4b79-863b-6a619b9dd9dc.png) ![percy_orb](https://user-images.githubusercontent.com/111712756/213612622-11be4083-cdaf-4f79-900d-3219c7a4c01b.png) |
 
[Beatmap: P*Light - SAtAN](https://osu.ppy.sh/beatmapsets/1358905#mania/2811967)

> I'd be more interested to know how stable behaves with very short holds such that the hold's "start" or "head" are overlapping – are they front-to-back as per normal notes?

I surrounded the textures with a red border for preview 
![noteT](https://user-images.githubusercontent.com/111712756/213608458-d5a0939c-15cc-40fb-8cb5-51d208ee2eaf.png)
![noteT2](https://user-images.githubusercontent.com/111712756/213608461-1c4d01d6-9765-4033-a498-d4961445c9c5.png)
| Hold end | Lazer | Stable |
| :---: | :---: | :---: |
| **Tail** | ![overlap-lazer](https://user-images.githubusercontent.com/111712756/213610535-22358fac-4e89-4f83-9ae2-62da63aad1f0.png) | ![overlap-stable](https://user-images.githubusercontent.com/111712756/213608027-159f3019-0099-415a-bc81-018e73c29dbc.png) |
| **Non-overlapping tail** | ![overlap_non-lazer](https://user-images.githubusercontent.com/111712756/213610749-c144cb08-5b4f-4848-950f-93c165aab2fd.png) | ![overlap_non-stable](https://user-images.githubusercontent.com/111712756/213608093-a07f1d73-dcd7-4db9-afdb-67676ce5a00d.png) |

**Comment by Adrriii** [comment]
Is there something planned to address this issue ? Because trying to counter this bug leads to weird display issues at the moment.

| In-game | LN Tail used |
|-----------|--------------|
| 1. When using a small LN tail, the body still goes halfway through : |
| ![image](https://github.com/ppy/osu/assets/16978299/5632b63e-b59a-4ac1-afd6-f3ef3b84800a) | ![tail](https://github.com/ppy/osu/assets/16978299/c2687e9d-4e40-4a12-bdc5-b35f49920dbc) |
| 2. Applying a black background on the LN tail overlaps the following notes : 
| ![image](https://github.com/ppy/osu/assets/16978299/998e3880-d5f0-48df-b45d-50916ef576ae)  | ![tail](https://github.com/ppy/osu/assets/16978299/a086c11b-1b29-40e8-b66b-b2bda13cb965) |
| 3. Reducing the black background creates an inconsistent white artefact at the top : |
| ![image](https://github.com/ppy/osu/assets/16978299/a1270ae4-b1c2-460e-8691-09fb344bbb52) | ![tail](https://github.com/ppy/osu/assets/16978299/7a4ca71d-a625-4d7a-9b8d-a27d4d9880f2) |

Out of the three solutions, 3 seems to be the most playable. However, the best case scenario would be to have the possibility of doing the following while using the first LN tail as some "pre-tail" asset instead: 
![image](https://github.com/ppy/osu/assets/16978299/ace20654-3f1b-4eab-abb6-f52797fd15c1)

**Comment by peppy** [comment]
> Is there something planned to address this issue?

No, because this issue is still open. If something was planned you would se something here.

Please do NOT work around this with skins and then come back when we break those skins complaining.

A setting is likely required at a skin level.

**Comment by Adrriii** [comment]
A skin setting is the best we could ask for regarding this matter (especially if it allows to have TopCenter origin, preventing overlaps altogether). We only come up with hacks to be able to play better in the meantime, it's not an end in itself. Thanks for the reply

**Comment by Adrriii** [comment]
> > Is there something planned to address this issue?
> 
> No, because this issue is still open. If something was planned you would se something here.
> 
> Please do NOT work around this with skins and then come back when we break those skins complaining.
> 
> A setting is likely required at a skin level.

Hi, sorry for digging up this issue, but at the moment some "skin workarounds" are still required to somewhat correctly represent release timing for skins with a complex LN Tail. It wouldn't be too much of an issue if such workarounds completely fixed the gameplay but it unfortunately is not the case at the moment. I believe progress was last made at https://github.com/ppy/osu/pull/27726 but don't quote me on that
